http-client/src/main/java/io/airlift/http/client/spnego/SpnegoAuthentication.java

```diff
@@ -193,6 +170,66 @@ public boolean matches(String type, URI uri, String realm)
         return NEGOTIATE.equalsIgnoreCase(type);
     }
 
+    private Session getCredential()
+            throws LoginException, GSSException
+    {
+        synchronized (this) {
+            checkState(!shutdown, "SpnegoAuthentication is already stopped");
+
+            if (clientSession == null || clientSession.getClientCredential().getRemainingLifetime() < MIN_CREDENTIAL_LIFE_TIME.getValue(TimeUnit.SECONDS)) {
+                if (clientSession != null) {
+                    clientSession.logout();
```

Comment:
One minor issue for this is that an outstanding request may fail if the clientSession object is being used in the middle while it's being logged out here. To prevent that, we may need to do reference counting. Otherwise, we are good.